File: String/longestSubstringWithoutRepitition.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        length = 0
        string_list = list(s)
        char_list = []
        
        for i in range(len(s)):
            curr_char = string_list[i]
            if curr_char in char_list:
                char_list = char_list[char_list.index(curr_char)+1:len(char_list)]
            char_list.append(curr_char)
            if len(char_list)>length:
                length = len(char_list)
        return length 

File: String/longestSubstringWithoutRepitition.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        vector<int> dictionary(256,-1); //initialize the vector of length 256 all initialized to -1
        int start = -1, max_length=0;
        
        for (int i=0; i<s.length(); i++){
            if(dictionary[s[i]]>start) //If there is repitition, then update the start to the index of the repitited char
                start = dictionary[s[i]];
            dictionary[s[i]]= i; // else update the vector with the char index
            max_length = max(max_length, i-start);
        }
        return max_length;
    }
};

//Solution learnt from @lightmark on leetcode
